This issue is described here: https://github.com/openwrt/packages/pull/8685
and has not been solved as of 1.70.0 release.

--- a/tools/build/src/tools/common.jam
+++ b/tools/build/src/tools/common.jam
@@ -973,18 +973,6 @@
         }
     }
 
-    # From GCC 5, versioning changes and minor becomes patch
-    if $(tag) = gcc && [ numbers.less 4 $(version[1]) ]
-    {
-        version = $(version[1]) ;
-    }
-
-    # Ditto, from Clang 4
-    if ( $(tag) = clang || $(tag) = clangw ) && [ numbers.less 3 $(version[1]) ]
-    {
-        version = $(version[1]) ;
-    }
-
     # On intel, version is not added, because it does not matter and it is the
     # version of vc used as backend that matters. Ideally, we should encode the
     # backend version but that would break compatibility with V1.
